src/common/factory/emailDeliveryIntegrator/emailDeliveryIntegrator.factory.ts

```diff
@@ -0,0 +1,35 @@
+import { SendgridEmailDeliveryIntegrator } from './sendgridEmailDeliveryIntegrator';
+import { EmailType } from '../../../email/core/email-type';
+import {
+  EmailKeyHost,
+  EmailRegistration,
+} from '../../../email/core/register-email';
+import { SenderOptions } from 'src/email/core/sender-options';
+
+export interface IEmailDeliveryIntegrator {
+  sendEmail(
+    template: EmailType<EmailRegistration & EmailKeyHost>,
+    emailRegistration: EmailRegistration,
+    to: string,
+    locals: Record<string, string>,
+    options?: SenderOptions,
+    subject?: string,
+  ): Promise<void>;
+}
+
+export default class EmailDeliveryIntegratorFactory {
```

Comment:
I've seen this done as a custom module in Nest. If it picks your interest, I can provide an example!

Reply:
@frankmangone Si podes mandarme ejemplo genial :)

Reply:
@FloPereira Capaz podés pedir acceso a este repo: https://bitbucket.org/spaceuy/pg-backend/src/main/

Ahí lo armé de esta manera que te digo, justamente para tener algo de referencia! 😄